content/learning-paths/smartphones-and-mobile/using_unity_machine_learning_agents/08-Training-our-ML-agents.md

```diff
@@ -0,0 +1,540 @@
+
+* Python - the ML Agents framework has been written in Python. [https://www.python.org](https://www.python.org/)
+
+* Python Pip - the package installer for Python. [https://pypi.org/project/pip](https://pypi.org/project/pip/)
+
+* PyTorch - a machine learning framework which ML Agents Toolkit wraps for Unity. [https://pytorch.org](https://pytorch.org/)
+
+* TensorFlow - Machine learning library [https://www.tensorflow.org](https://www.tensorflow.org/)
+
+* TensorBoard - visualization tool for Tensorflow [https://www.tensorflow.org/tensorboard](https://www.tensorflow.org/tensorboard)
+
+* ML Agents - part of the ML Agents Toolkit from Unity which provides imitation and reinforcement learning [https://pypi.org/project/mlagents](https://pypi.org/project/mlagents/). The Unity ML Agents toolkit github is [https://github.com/Unity-Technologies/ml-agents](https://github.com/Unity-Technologies/ml-agents)
+
+In the next section we'll go through the steps to get all of the above installed.
+
+## Installing Python, Pip and the required packages
+
+### Installing Python
+
+We recommend using Python 3.7.9 as future versions produced errors on some platforms when installing PyTorch.
+
+In a command or terminal prompt (such as Power Shell):
+
+1. Install Python 3.7.9 (or a later 3.7), and pip 20.1.1 or later. The official python installer can be found on:
+https://www.python.org/downloads/
+
+1. Ensure pip is installed by following the instructions for your operating system on this page: https://pip.pypa.io/en/stable/installation/
+
+### Installing PyTorch and ML Agents
+
+_Note 1._ If you are using a command prompt or power shell you may need to run with administrator permissons.
+
+_Note 2._ In Windows you may need to enable [LongPath support](https://pip.pypa.io/warnings/enable-long-paths).
+
+1. Open a command prompt (or similar such as Visual Studio Code Terminal)
+
+1. Change directory to your copy of the Unity project
+
+    ```
+    cd MLAgentsWorkshopSupportingFiles
+    ```
+
+1. If you list the content of the directory, you should see the Unity project folders
+
+    ```
+    dir /b
+    ```
+
+    ```
+    Assets
+    MixamoNoLicense.png
+    Packages
+    ProjectSettings
+    ```
+
+1. Create a python virtual environment. 
+
+    During the following you may need to use `python3` instead of `python`, depending on your installation:
+
+    ```
+    md python-envs
+    ```
+
+    ```
+    python -m venv .\python-envs\mlagents-r18-env
+    ```
+
+    If you have problems installing PyTorch (step 4 below), try deleting "mlagents-r18-env" directory and retry with the py tool:
+
+    ```
+    py -3.7 -m venv .\python-envs\mlagents-r18-env
+    ```
+
+1. Activate the virtual environment:
+
+    ```
+    .\python-envs\mlagents-r18-env\Scripts\activate
+    ```
+
+    If activated successfully, your prompt will be prefixed by "(mlagents-r18-env)"
+
+1. Check your active python version:
+
+    ```
+    python --version
+    ```
```

Comment:
We're asking what version, but we should say what we should get.  Maybe "Check your active python version is the one you installed."?